### ISSUE TITLE
[CI] Add linting jobs

### ISSUE THREAD

**Issue by devkabiir** [exercism/dart#307]
The current CI (GitHub Actions) only runs `configlet` based linting and dart tests. Linting for dart should also be part of the CI.

- Add dart analyzer linting job
- Add dart formatting lint job (The track currently uses `120` characters line width)

**Comment by Stargator** [comment]
@devkabiir Can the CI modify the code in order to fo the dart formatting? Or are you just saying to do a formatting check?

**Comment by devkabiir** [comment]
Related #296

Making changes requires an access token for most CI. I'm also not comfortable with any kind of CI to make changes.

My proposal is only for linting and checking for formatting.

**Comment by Stargator** [comment]
Parts of this are touched on in #227 which handles having a workflow or CI.

@devkabiir Is there a difference between the two you could make?

**Comment by devkabiir** [comment]
I suppose this one was to add the CI jobs, and the other one was to select the lint rules, other than the default ones enabled by dart analyzer.

**Comment by Stargator** [comment]
@devkabiir [The Test workflow does check the formatting](https://github.com/exercism/dart/blob/main/.github/workflows/test.yml#L34) unless you want to refactor the formatting out of the Test workflow and into its own Workflow.

Since both workflows are executed when a Dart file or analysis_options.yaml file is changed, then they would always execute on the same PRs. Combining them just reduces our use of GitLab resources.

**Comment by devkabiir** [comment]
No, I don't think separate workflows add any value. However, we should run `dart analyze .`, `dart format -l 120 --set-exit-if-changed .` per exercise as well as repository level. That was the intention as far as I can remember.
